src/main/java/net/syscon/elite/repository/jpa/model/CourtEvent.java

```diff
@@ -115,19 +126,48 @@ public void setEventDateTime(final LocalDateTime dateTime) {
         this.startTime = dateTime;
     }
 
-    @PrePersist
-    private void carryOverPreExistingActiveChargesOnCreation() {
-        if (offenderCourtCase != null) {
-            log.debug("Adding pre-existing active charges to new court event for court case '{}", offenderCourtCase.getId());
+    private void add(final CourtEventCharge charge) {
+        charges.add(charge);
+    }
 
-            offenderCourtCase.getCharges().stream().filter(OffenderCharge::isActive).forEach(this::addPreExisting);
+    static class OnCreate {
+        /**
+         * On court event creation on NOMIS existing active charges are applied to the new court event if associated with a court case.
+         */
+        @PrePersist
+        void applyActiveCourtCaseChargesFor(final CourtEvent event) {
+            event.getOffenderCourtCase().ifPresent(courtCase -> {
```

Comment:
So interestingly this will probably cause a SELECT assuming this is lazy loaded - which I'm sure is OK